astropy/config/paths.py

```diff
@@ -19,6 +20,46 @@
 ]
 
 
+class _DirectoryType(Enum):
+    CONFIG = auto()
+    CACHE = auto()
+
+
+def _get_dir_path(rootname: str, dirtype: _DirectoryType) -> Path:
+    match dirtype:
+        case _DirectoryType.CONFIG:
```

Comment:
Seems a little more complicated than what I had in mind. It is very unlikely we will ever expand this, so I don't think we need to make it so flexible.

Reply:
I don't mean it to become more flexible, though is it a bad thing if it merely emerges as such ? this `match/case` is just a fancy if/elif/else, and the reason I'm combining it with an enum is that it makes the dispatching logic checkable by a type-checker (see https://github.com/astropy/astropy/issues/16603). Admittedly it's not required, but should I throw it away after the work is done ?

Reply:
Maybe I'll let @nstarman have a look then since he initially requested such refactoring and is very into typing stuff. Thanks for the clarification!

Reply:
I'm a bit with @pllim here, and would just have the call signature be `_get_dir_path(fallback, cls, rootname="astropy")` so that one can avoid the whole matching/ifelsing (fine to omit error checking since it is just used internally).

Reply:
I agree that the Enum + match is cool, but unnecessary in this case. An `if-else` would be best here unless we plan to add more options to the enum. The flag can be a `bool`, which is equally good type-wise for 2 options.

Reply:
Alright, by popular demand, here's a simpler impl (enum and match/case-free)